syntax = "proto2";
package pb;
import "room.proto";
import "value.proto";

message iCreateRoomREQ
{
    optional CreateRoomREQ  req = 1;
    optional uint32         roomid = 2;
    optional uint64         uid = 3;            // 创建者id
    optional string         room_service_name = 4;
}

message iCreateRoomRSP
{
    optional int32          code = 1;       // 0 成功
    optional uint32         roomid = 2;
    optional uint64         uid = 3;
    optional uint32         clubid = 4;
    optional string         room_name = 5;
}

message iRegisterRoomServer
{
    optional uint32 handle = 1;
    optional string name = 2;
}

message iRoomServerHeartBeat
{
    optional string name = 1;
}

message iRoomBriefReport
{
    optional uint32     handle = 1;
    optional RoomBrief  room = 2;
}

message iRoomOver
{
    optional uint32 clubid = 1;
    optional uint32 roomid = 2;
}

message iEnterRoom
{
    optional uint32     clubid = 1;
    optional uint64     uid = 2;
    optional uint32     handle = 3;     // agent_handle
    optional string     ip = 4;
}

message iAgentRoomStatus
{
    optional bool       enter = 1;          // 是否是进入
    optional uint32     roomid = 2;     
    optional uint32     handle = 3;         // room_handle 
    optional uint32     clubid = 4;
}

message iReloadUserInfo
{
    optional uint64     uid = 1;
}

message iBaseUserInfo
{
    optional uint64     uid = 1;
    optional string     name = 2;
    optional string     icon = 3;
    optional uint64     last_login_time = 4;
}

message iLoadRoomListUserREQ
{
    optional uint64     uid = 1;
}

message iLoadRoomListUserRSP
{
    optional iBaseUserInfo  info = 1;
}

message iLoadClubUserREQ
{
    optional uint64     uid = 1;
}

message iLoadClubUserRSP
{
    optional iBaseUserInfo  info = 1;
    repeated uint32         clubids = 2;       // 玩家所在的俱乐部列表   
}

enum ValueOP
{
    VALUEOP_INVALID = 0;
    VALUEOP_GET = 1;
    VALUEOP_DELETE = 2;
    VALUEOP_CHANGE = 3;
}

message iChangeValueREQ
{
    optional ValueType  value_type = 1;
    optional ValueKey   key = 2;
    optional ValueOP    op = 3[default = VALUEOP_GET];
    optional int64      change = 4[default = 0];
    optional string     type = 5;
    optional string     attach = 6;
}

message iChangeValueRSP
{
    optional int32      code = 1;   // 0成功
    optional int64      value = 2;
}

message iClubRoomNum
{
    optional uint32     clubid = 1;
    optional uint32     num = 2;
}

message iGetClubRoomNumREQ
{
    repeated uint32     clubids = 1;
}

message iGetClubRoomNumRSP
{
    repeated iClubRoomNum   items = 1;
}

message iRoomRecord
{
    optional string         room_set_id = 1;
    optional RoomBrief      brief = 2;
    optional uint64         ownerid = 3;
}

message iGameUserRecord
{
    optional uint32         clubid = 1;
    optional uint64         uid = 2;
    optional uint64         rake = 3;               // 抽水
    optional int64          chips = 4;              // 输赢筹码
}

message iGameRecord
{
    optional string             room_set_id = 1;
    optional string             game_id = 2;
    optional uint64             game_time = 3;
    repeated iGameUserRecord    user_records = 4;
}
